File: spring/oidc/oidc-example/src/main/java/com/example/oidcexample/HelloController.java
package com.example.oidcexample;

import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/")
@Slf4j
public class HelloController {

    @GetMapping
    public String index() {
        final Authentication authn = SecurityContextHolder.getContext().getAuthentication();
        log.info("Authentication: {}", authn);
        return "Hello " + authn.getName() + "!";
    }
}

File: spring/with-vite/src/main/kotlin/com/example/with_vite/HelloController.kt
package com.example.with_vite

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/hello")
class HelloController {
    @GetMapping
    fun index(model: Model): String{
        model.addAttribute("name", "Vite")
        return "hello/index"
    }
}
